lib/models/fee_options/fee_option.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:breez_sdk/bridge_generated.dart';
+import 'package:breez_translations/generated/breez_translations.dart';
+
+enum ProcessingSpeed {
+  economy(Duration(minutes: 60)),
+  regular(Duration(minutes: 30)),
+  priority(Duration(minutes: 10));
+
+  const ProcessingSpeed(this.waitingTime);
+  final Duration waitingTime;
+}
+
+abstract class FeeOption {
+  final int txFeeSat;
+  final ProcessingSpeed processingSpeed;
+  final int satPerVbyte;
+
+  FeeOption({
+    required this.txFeeSat,
+    required this.processingSpeed,
+    required this.satPerVbyte,
+  });
+
+  String getDisplayName(BreezTranslations texts) {
+    switch (processingSpeed) {
+      case ProcessingSpeed.economy:
+        return texts.fee_chooser_option_economy;
+      case ProcessingSpeed.regular:
+        return texts.fee_chooser_option_regular;
+      case ProcessingSpeed.priority:
+        return texts.fee_chooser_option_priority;
+    }
+  }
+
+  bool isAffordable({required int balance, required int amountSat, bool? isMaxValue});
+}
+
+class ReverseSwapFeeOption extends FeeOption {
+  final PrepareOnchainPaymentResponse pairInfo;
+
+  ReverseSwapFeeOption({
+    required this.pairInfo,
+    required super.txFeeSat,
+    required super.processingSpeed,
+    required super.satPerVbyte,
+  });
+
+  @override
+  bool isAffordable({required int balance, required int amountSat, bool? isMaxValue}) {
+    if (isMaxValue == true) {
+      return (amountSat + boltzServiceFee(amountSat) + pairInfo.feesClaim) > 0;
```

Comment:
Isn't this always true?

Reply:
I guess I wanted to write balance - (fees) > 0 but it's the same with the statement below.

I wasn't able to give this much attention while working on notifications. I'm reviewing & testing this whole thing again with fresh eyes.